Build failures look real:
```
======================================================================
FAIL: corehq.apps.app_manager.tests.test_case_list_form:CaseListFormSuiteTests.test_case_list_form_parent_child_submodule_advanced
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/test_case_list_form.py", line 265, in test_case_list_form_parent_child_submodule_advanced
    self.assertXmlEqual(self.get_xml('case-list-form-suite-parent-child-submodule-advanced'), factory.app.create_suite())
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/util.py", line 32, in assertXmlEqual
    _check_shared(expected, actual, LXMLOutputChecker(), "xml")
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/util.py", line 126, in _check_shared
    raise AssertionError(message)
AssertionError: XML mismatch
--- want.xml
+++ got.xml
@@ -254,7 +254,6 @@
         <command value="'m2'"/>
         <datum id="case_id_load_house_0" value="instance('commcaresession')/session/data/case_id_load_house_0"/>
         <command value="'m3'"/>
-        <datum id="case_id_load_person_0" value="instance('commcaresession')/session/data/case_id_new_person_0"/>
       </create>
       <create if="count(instance('commcaresession')/session/data/return_to) = 1 and instance('commcaresession')/session/data/return_to = 'm3' and count(instance('casedb')/casedb/case[@case_id=instance('commcaresession')/session/data/case_id_new_person_0]) = 0">
         <command value="'m2'"/>

======================================================================
FAIL: corehq.apps.app_manager.tests.test_form_workflow:TestFormWorkflow.test_form_links_submodule
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/test_form_workflow.py", line 321, in test_form_links_submodule
    self.assertXmlPartialEqual(self.get_xml('form_link_submodule'), factory.app.create_suite(), "./entry")
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/util.py", line 26, in assertXmlPartialEqual
    self.assertXmlEqual(expected, actual, normalize=False)
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/util.py", line 32, in assertXmlEqual
    _check_shared(expected, actual, LXMLOutputChecker(), "xml")
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/util.py", line 126, in _check_shared
    raise AssertionError(message)
AssertionError: XML mismatch
--- want.xml
+++ got.xml
@@ -29,8 +29,8 @@
         <command value="'m1'"/>
         <datum id="case_id" value="instance('commcaresession')/session/data/case_id"/>
         <datum id="case_id_new_visit_0" value="uuid()"/>
+        <command value="'m1-f1'"/>
         <datum id="case_id_load_visit_0" value="instance('commcaresession')/session/data/case_id_load_visit_0"/>
-        <command value="'m1-f1'"/>
       </create>
     </stack>
   </entry>
```

--- a/corehq/apps/app_manager/suite_xml/post_process/workflow.py
+++ b/corehq/apps/app_manager/suite_xml/post_process/workflow.py
@@ -47,7 +47,7 @@ class WorkflowHelper(PostProcessor):
 
                 self.create_workflow_stack(form_command, stack_frames)
 
-    def get_frame_children(self, target_form, module_only=False):
+    def get_frame_children(self, target_form, module_only=False, include_target_root=False):
         """
         For a form return the list of stack frame children that are required
         to navigate to that form.
@@ -84,7 +84,7 @@ class WorkflowHelper(PostProcessor):
         else:
             datums_list = module_datums.values()  # [ [datums for f0], [datums for f1], ...]
             root_module = target_form.get_module().root_module
-            if root_module:
+            if root_module and include_target_root:
                 datums_list = datums_list + self.get_module_datums(id_strings.menu_id(root_module)).values()
 
         common_datums = commonprefix(datums_list)
@@ -269,7 +269,7 @@ class EndOfFormNavigationWorkflow(object):
                 frame_children = frame_children_for_module(root_module)
                 return StackFrameMeta(xpath, frame_children)
             elif form_workflow == WORKFLOW_PREVIOUS:
-                frame_children = self.helper.get_frame_children(form)
+                frame_children = self.helper.get_frame_children(form, include_target_root=True)
 
                 # since we want to go the 'previous' screen we need to drop the last
                 # datum

